pyccel/codegen/printing/ccode.py

```diff
@@ -666,31 +666,54 @@ def _print_IndexedElement(self, expr):
                 #managing the Slice input
                 for i , ind in enumerate(inds):
                     if isinstance(ind, Slice):
-                        #setting the slice start and end to their correct value if none is provided
-                        start = ind.start
-                        #handling the negative indexes for a slice object
-                        if isinstance(start, PyccelUnarySub) and \
-                                isinstance(start.args[0], LiteralInteger):
-                            start = PyccelMinus(base_shape[i], start.args[0])
-                        end = ind.end
-                        if isinstance(end, PyccelUnarySub) and \
-                                isinstance(end.args[0], LiteralInteger):
-                            end = PyccelMinus(base_shape[i], end.args[0])
-                        if ind.start is None:
-                            start = 0
-                        if ind.end is None:
-                            end = base.shape[i]
-                        inds[i] = Slice(start, end)
-                    else:
-                        #setting the Slice start and end to their correct value when try to get a view with scalar index
-                        inds[i] = Slice(ind, PyccelAdd(ind, LiteralInteger(1)))
+                        inds[i] = self._slice_elements_managements(ind, PyccelArraySize(base, i),
+                            allow_negative_indexes)
                 inds = [self._print(i) for i in inds]
                 return "array_slicing(%s, %s)" % (base_name, ", ".join(inds))
             inds = [self._print(i) for i in inds]
         else:
             raise NotImplementedError(expr)
         return "%s.%s[get_index(%s, %s)]" % (base_name, dtype, base_name, ", ".join(inds))
 
+    def _slice_elements_managements(self, _slice, shape, allow_negative_index):
```

Comment:
I have a few comments about this function:

1. Since the `self` argument is not used, we should make this a `@staticmethod`
2. Even though this is a private method, it is important enough to deserve a **docstring**
3. The function's name is not sufficiently clear. Given that a new `Slice` object is created with explicit values of `(start, stop, step)`, maybe something like `_new_slice_with_processed_arguments`?
4. The `shape` argument does not represent the proper shape of the array (which is a tuple), but only the number of elements along the given index; so we could call it `array_size` to avoid confusion.

Reply:
is this a decent docstring? 
        """ Create new slice with informations collected from old slice and decorators 
            
        Parameters
        ----------
            _slice : Slice
                slice needed to collect (start, stop, step)
            array_size : PyccelArraySize
                call to function size()
            allow_negative_index = Bool
                True when the decorator allow_negative_index is present
        Returns
        -------
            Slice
        """

Reply:
`allow_negative_index = Bool`->`allow_negative_index : Bool`

Reply:
Yes, this works. Thanks